plugins/modules/proxmox_snap_info.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, Jeffrey van Pelt (@Thulium-Drake) <jeff@vanpelt.one>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_snap_info
+short_description: Snapshot management of instances in Proxmox VE cluster
+version_added: 6.3.0
+description:
+  - Lists all snapshots from instances in Proxmox VE cluster.
+  - to be optionally combined with older_than
+  - to be optionally combine with snapname (finds snapshot starting with the snapname)
+  - Supports both KVM and LXC, OpenVZ has not been tested, as it is no longer supported on Proxmox VE.
+options:
+  hostname:
+    description:
+      - The instance name.
+    type: str
+  vmid:
+    description:
+      - The instance ID.
+      - If not set, will be fetched from PromoxAPI based on the hostname.
+    type: str
+  timeout:
+    description:
+      - Timeout for operations.
+    default: 30
+    type: int
+  snapname:
+    description:
+      - Beginning of the snapshot name proxmox_snap_info takes care of.
+    type: str
+  older_than:
+    description:
+      - Minimum age of backup to be listed by proxmox_snap_info in days
+    default: 0
+    type: int
+
+notes:
+  - Requires proxmoxer and requests modules on host. These modules can be installed with pip.
+requirements: [ "proxmoxer", "python >= 2.7", "requests" ]
+author: Jeffrey van Pelt (@Thulium-Drake)
```

Comment:
If you copied this mostly from a module @Thulium-Drake created, keeping him as the author is nice, but you should also add yourself.